frontends/web/src/components/language/language.tsx

```diff
@@ -80,10 +80,15 @@ const LanguageSwitch: FunctionComponent<TLanguageSwitchProps> = ({ languages })
   const [selectedIndex, setSelectedIndex] = useState<number>(getSelectedIndex(allLanguages, i18n));
   const [activeDialog, setActiveDialog] = useState<boolean>(false);
 
+  const dialogRef = useRef<Dialog>() as React.MutableRefObject<Dialog>;
+
   const changeLanguage = (langCode: TActiveLanguageCodes, index: number) => {
     setSelectedIndex(index);
-    setActiveDialog(false);
     i18n.changeLanguage(langCode);
+    dialogRef.current.deactivate();
+    setTimeout(() => {
+      setActiveDialog(false);
+    }, 350);
```

Comment:
This is needed to preserve the animation of dismissing the modal after clicking a language:

1. We need to trigger `deactivate()` which shows the animation & deactivates the dialog. 

With the way `Dialog` works right now, I haven't found a "better" way to trigger the animation from a parent component, such as `language.tsx` here.

2. And after 350ms (after the animation has finished), we can set `activeDialog` to `false` 

We need to give a 350ms delay here before setting `activeDialog` to `false`, because otherwise the component won't have a chance to animate. It'd just be dismissed instantly without any animation due to conditional rendering `activeDialog && <Dialog...`, and this isn't what we want.

I can see this being refactored, but we first need to start from the `Dialog` component, probably utilising `React.createContext` so we don't have to conditional render `<Dialog/>` on each page where it's being used thus reducing redundancy.